pkg/reconciler/revision/revision_test.go

```diff
@@ -488,118 +492,64 @@ func TestNoQueueSidecarImageUpdateFail(t *testing.T) {
-			revClient := servingClient.ServingV1().Revisions(rev.Namespace)
+	ctx, cancel := context.WithCancel(ctx)
+	grp := errgroup.Group{}
 
-			h := NewHooks()
+	servingClient := fakeservingclient.Get(ctx)
 
-			h.OnUpdate(&servingClient.Fake, "revisions", test.callback(t))
+	rev := testRevision(getPodSpec())
+	revClient := servingClient.ServingV1().Revisions(rev.Namespace)
 
-			waitInformers, err := controller.RunInformers(ctx.Done(), informers...)
-			if err != nil {
-				t.Fatal("Failed to start informers:", err)
-			}
-			defer func() {
-				cancel()
-				if err := grp.Wait(); err != nil {
-					t.Errorf("Wait() = %v", err)
-				}
-				waitInformers()
-			}()
+	waitInformers, err := controller.RunInformers(ctx.Done(), informers...)
+	if err != nil {
+		t.Fatal("Failed to start informers:", err)
+	}
+	defer func() {
+		cancel()
+		if err := grp.Wait(); err != nil {
+			t.Errorf("Wait() = %v", err)
+		}
+		waitInformers()
+	}()
 
-			if err := watcher.Start(ctx.Done()); err != nil {
-				t.Fatal("Failed to start configuration manager:", err)
-			}
+	if err := watcher.Start(ctx.Done()); err != nil {
+		t.Fatal("Failed to start configuration manager:", err)
+	}
 
-			grp.Go(func() error { return ctrl.Run(1, ctx.Done()) })
+	grp.Go(func() error { return ctrl.Run(1, ctx.Done()) })
 
-			revClient.Create(rev)
-			revL := fakerevisioninformer.Get(ctx).Lister()
-			if err := wait.PollImmediate(10*time.Millisecond, 5*time.Second, func() (bool, error) {
-				l, err := revL.List(labels.Everything())
-				// We only create a single revision.
-				return len(l) > 0, err
-			}); err != nil {
-				t.Fatal("Failed to see Revision propagation:", err)
-			}
-			t.Log("Seen revision propagation")
+	revClient.Create(rev)
+	revL := fakerevisioninformer.Get(ctx).Lister()
+	if err := wait.PollImmediate(10*time.Millisecond, 5*time.Second, func() (bool, error) {
+		l, err := revL.List(labels.Everything())
+		// We only create a single revision, but make sure it is reconciled.
+		return len(l) > 0 && !cmp.Equal(l[0], rev), err
+	}); err != nil {
+		t.Fatal("Failed to see Revision propagation:", err)
+	}
+	t.Log("Seen revision propagation")
 
-			watcher.OnChange(test.configMapToUpdate)
+	watcher.OnChange(&corev1.ConfigMap{
+		ObjectMeta: metav1.ObjectMeta{
+			Namespace: system.Namespace(),
+			Name:      metrics.ConfigMapName(),
+		},
+		Data: map[string]string{
+			"logging.enable-var-log-collection": "true",
+			"logging.revision-url-template":     "http://log-here.test.com?filter=${REVISION_UID}",
+		},
+	})
 
-			if err := h.WaitForHooks(3 * time.Second); err != nil {
-				t.Error("Global Resync Failed:", err)
-			}
-		})
+	const expected = "http://log-here.test.com?filter="
+	if err := wait.PollImmediate(50*time.Millisecond, 5*time.Second, func() (bool, error) {
+		l, _ := revL.List(labels.Everything())
+		// We are guaranteed to have at least 1 element.
+		got := l[0].Status.LogURL
+		return strings.HasPrefix(got, expected), nil
```

Comment:
We could now even tighten this up :thinking:. We can use the revision's UID (which I suppose is empty) and make an exact check. Not super necessary though.

Reply:
I don't see whynot.